src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -147,18 +151,53 @@ const submitEncryptModeForm: ControllerHandler<
 
   // Check captcha
   if (form.hasCaptcha) {
-    const captchaResult = await CaptchaService.verifyCaptchaResponse(
-      req.query.captchaResponse,
-      getRequestIp(req),
-    )
-    if (captchaResult.isErr()) {
+    const featureFlagsListResult = await FeatureFlagService.getEnabledFlags()
+    // Feature flag to control turnstile captcha rollout
+    // defaults to false
+    // todo: remove after full rollout
+    let enableTurnstileFeatureFlag = false
+    if (featureFlagsListResult.isErr()) {
       logger.error({
-        message: 'Error while verifying captcha',
+        message: 'Error occurred whilst retrieving enabled feature flags',
         meta: logMeta,
-        error: captchaResult.error,
+        error: featureFlagsListResult.error,
       })
-      const { errorMessage, statusCode } = mapRouteError(captchaResult.error)
-      return res.status(statusCode).json({ message: errorMessage })
+    } else {
+      enableTurnstileFeatureFlag = featureFlagsListResult.value.includes(
+        featureFlags.turnstile,
+      )
+    }
```

Comment:
/nit: Is this code straight duplicate from `email-submission.controller.ts`? Any way to have it in some shared utility?

Reply:
Both are written quite differently and will require significant refactoring. I can take a look at how we can restructure both controllers

Reply:
Oh? Per the highlight (line 154 to 169), I meant just the code to gather and compute `featureFlagsListResult`. Isn't that a straight copy?

here
```typescript
    const featureFlagsListResult = await FeatureFlagService.getEnabledFlags()
    // Feature flag to control turnstile captcha rollout
    // defaults to false
    // todo: remove after full rollout
    let enableTurnstileFeatureFlag = false
    if (featureFlagsListResult.isErr()) {
      logger.error({
        message: 'Error occurred whilst retrieving enabled feature flags',
        meta: logMeta,
        error: featureFlagsListResult.error,
      })
    } else {
      enableTurnstileFeatureFlag = featureFlagsListResult.value.includes(
        featureFlags.turnstile,
      )
    }
```

In `email-submission.controller.ts`
```typescript
  const featureFlagsListResult = await FeatureFlagService.getEnabledFlags()
  // Feature flag to control turnstile captcha rollout
  // defaults to false
  // todo: remove after full rollout
  let enableTurnstileFeatureFlag = false
  if (featureFlagsListResult.isErr()) {
    logger.error({
      message: 'Error occurred whilst retrieving enabled feature flags',
      meta: logMeta,
      error: featureFlagsListResult.error,
    })
  } else {
    enableTurnstileFeatureFlag = featureFlagsListResult.value.includes(
      featureFlags.turnstile,
    )
  }
```

Reply:
Hopefully we could put that in a shared function such that the code in both submission handlers could be reduced to a minimum
```typescript
const enableTurnstileFeatureFlag = await FeatureFlagService.hasFeatureFlag(featureFlags.turnstile)
```

Actually, don't we already have a function like that?

Reply:
But yeah, so just to be super clear: full refactor of the encrypt and email controller is indeed be a completely separate task; definitely out of scope for this PR, and I didn't doing mean that with my comment 😅

Reply:
I've some plans to refactor them in d2a8816f1, c18d8bae6, and 52cf50aac